skops/card/_model_card.py

```diff
@@ -1410,13 +1425,18 @@ def save(self, path: str | Path) -> None:
         path: str, or Path
             Filepath to save your card.
 
+        plot_path: str
+            Filepath to save the plots.
+
         Notes
         -----
         The keys in model card metadata can be seen `here
         <https://huggingface.co/docs/hub/models-cards#model-card-metadata>`__.
         """
         with open(path, "w", encoding="utf-8") as f:
-            f.write("\n".join(self._generate_card()))
+            if not isinstance(path, Path):
```

Comment:
There is a bit of redundancy here: You cast `path` to a `Path` object here but `_generate_card` and `_generate_content` both accept `str` and the latter does `Path(destination_path)`. So either you just pass `path` without conversion, or those methods should not accept `str` and don't perform a 2nd round of conversion.